ctapipe/core/tool.py

```diff
@@ -189,18 +209,40 @@ def initialize(self, argv=None):
         self.update_logging_config()
 
         if self.config_file is not None:
-            self.log.debug(f"Loading config from '{self.config_file}'")
+            self.log.info(f"Loading config from '{self.config_file}'")
             try:
-                self.load_config_file(self.config_file)
+                for config_file in self.config_file:
+                    self.load_config_file(config_file)
             except Exception as err:
-                raise ToolConfigurationError(f"Couldn't read config file: {err}")
+                raise ToolConfigurationError(
+                    f"Couldn't read config file: {err} ({type(err)})"
+                )
 
         # ensure command-line takes precedence over config file options:
         self.update_config(self.cli_config)
         self.update_logging_config()
 
         self.log.info(f"ctapipe version {self.version_string}")
 
+    def load_config_file(self, path: Union[str, pathlib.Path]):
+
+        path = pathlib.Path(path)
+
+        if path.suffix in [".yaml", ".yml"] and HAS_YAML:
+            # do our own YAML loading
+            with open(path, "r") as infile:
+                config = Config(yaml.safe_load(infile))
+            self.update_config(config)
```

Comment:
I think this overrides previous configs fully. Looking at the traitlets code for `load_config_file`, they do something like this instead:

```
config  = Config()
for config_file in self.config_files:
     config.merge(load(config_file))
self.update_config(config)
```

Reply:
We should test with two configs setting independent things.

Reply:
The main reason they merge a bunch together first was to be able to print an error when one fails, I think (at least that is what I understood from the code), but it would be good to check that this doesn't break anything.  The idea is that  a later config file *should* overwrite earlier settings, but it should not drop old configs if that is what you mean, i.e. that `update_config()` *replaces* rather than *updates* previous ones.

Reply:
I'll add a test in any case

Reply:
I feared that loading two config files with this:

```
{"a": 1}
```

```
{"b": 2}
```

results in just `{"b": 2}` if we do `update` instead of `merge`, but I didn't check

Reply:
This is the implementation of `update_config()` in Traitlets:
```py3
   def update_config(self, config):
        """Update config and load the new values"""
        # traitlets prior to 4.2 created a copy of self.config in order to trigger change events.
        # Some projects (IPython < 5) relied upon one side effect of this,
        # that self.config prior to update_config was not modified in-place.
        # For backward-compatibility, we must ensure that self.config
        # is a new object and not modified in-place,
        # but config consumers should not rely on this behavior.
        self.config = deepcopy(self.config)
        # load config
        self._load_config(config)
        # merge it into self.config
        self.config.merge(config)
```
So I think it is ok: "update" here means "load and merge"